### ISSUE TITLE
[a11y]: slider label is not used for slider input label

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#10172]
### Package

carbon-components

### Browser

Chrome

### Operating System

MacOS

### Package version

https://react.carbondesignsystem.com/?path=/story/components-slider--default

### Automated testing tool and ruleset

n/a

### Description

The label for the slider should be associated with both the thumb slider and the input.

Fully discussed in internal discussion https://ibm-studios.slack.com/archives/G01GCBCGTPV/p1631717241028100

### WCAG 2.1 Violation

3.3.2 Label or Instructions

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-slider--default

### Steps to reproduce

Inspect the code. The input's label is hard-coded by aria-label, instead of pointing to whatever value the designer has assigned for the text appearing as "Slider label"

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
